### PR TITLE
Add function to trim whitespaces from start/end of string

### PR DESCRIPTION
**Pull request by mkiael**

### CODE REVIEW

File: util/string_test.cpp [hunk 1]
```diff
@@ -25,5 +29,30 @@ int main() {
...
+        expect_eq(trim_start("word "), "word ");
+        expect_eq(trim_start("\r\n"), "");
+    });
+
+    etest::test("trim end", [] {
+        expect_eq(trim_end("abc "), "abc");
+        expect_eq(trim_end("53 \r\n"), "53");
+        expect_eq(trim_end("hello world!\t"), "hello world!");
+        expect_eq(trim_end(" word"), " word");
+        expect_eq(trim_start("\r\n"), "");
```
Comment (robinlinden) [inline util/string_test.cpp:45]: Should be calling `trim_end` and not `trim_start`?

File: util/string_test.cpp [hunk 2]
```diff
@@ -25,5 +29,30 @@ int main() {
...
+    });
+
+    etest::test("trim", [] {
+        expect_eq(trim("abc"), "abc");
+        expect_eq(trim_start("\t431"), "431");
```
Comment (robinlinden) [inline util/string_test.cpp:50]: Should be calling `trim` and not `trim_start`?

File: util/string_test.cpp [hunk 3]
```diff
@@ -25,5 +29,30 @@ int main() {
...
+        expect_eq(trim("53 \r\n"), "53");
+        expect_eq(trim("\t\thello world\n"), "hello world");
+        expect_eq(trim(" a b c d "), "a b c d");
+        expect_eq(trim_start("\r\n"), "");
```
Comment (robinlinden) [inline util/string_test.cpp:54]: Should be calling `trim` and not `trim_start`?

File: util/string.h [hunk 4]
```diff
@@ -24,6 +27,22 @@ constexpr bool no_case_compare(std::string_view a, std::string_view b) {
     return true;
 }
 
+inline std::string trim_start(std::string s) {
+    s.erase(begin(s), std::find_if(begin(s), end(s), [](auto ch) { return !std::isspace(ch); }));
```
Comment (robinlinden) [inline util/string.h:31]: You can replace this with https://en.cppreference.com/w/cpp/string/basic_string/erase2 and not have to deal with iterators at all. Awesome new thing in C++20!

File: util/string.h [hunk 5]
```diff
@@ -24,6 +27,22 @@ constexpr bool no_case_compare(std::string_view a, std::string_view b) {
...
+    return s;
+}
+
+inline std::string trim_end(std::string s) {
+    s.erase(std::find_if(rbegin(s), rend(s), [](auto ch) { return !std::isspace(ch); }).base(), end(s));
+    return s;
+}
+
+inline std::string trim(std::string s) {
+    s = trim_start(s);
```
Comment (robinlinden) [inline util/string.h:41]: You could move s into these.

**Review by robinlinden — CHANGES_REQUESTED** [review]

File: util/string_test.cpp (hunk 1 above)
Comment (mkiael) [reply util/string_test.cpp:45]: That was sloppy! Good catch!

File: util/string_test.cpp (hunk 2 above)
Comment (mkiael) [reply util/string_test.cpp:50]: Good catch!

File: util/string.h (hunk 4 above)
Comment (mkiael) [reply util/string.h:31]: Oh, nice. I'll fix!

**Review by robinlinden — APPROVED** [review]
Looks good!

**Pull request merged**